dev/io.openliberty.wsoc.internal_fat/fat/src/io/openliberty/wsoc/tests/BasicTest.java

```diff
@@ -216,6 +224,30 @@ public void testURIOverlappingWithHtml() throws Exception {
         this.verifyResponse(browser, "/basic/URIOverlapping.html", expectedInResponse);
     }
 
+    @Mode(TestMode.FULL)
+    // We expect to see an IllegalStateException due to invalid state when deploying endpoint.addsclosed
+    // We also expect to see java.io.IOException: Stream is closed when the request fails to process due to the IllegalStateException
+    @ExpectedFFDC(value = { "java.lang.IllegalStateException", "java.io.IOException" }, repeatAction = { EmptyAction.ID, JakartaEE9Action.ID })
+    // We allow the FFDC due to "Websocket request processed but provided upgrade header "null" does not match websocket" when upgrading request
+    @AllowedFFDC({ "java.lang.Exception" })
+    @Test
+    public void testAddEndpointOutisdeDeploymentPhase() throws Exception {
+        // New test for WebSocket 2.1 change to allow adding endpoints outside of initialization/deployment phase of web application https://github.com/jakartaee/websocket/issues/211
+        
+        // Call to open the connection to the servlet. Need to add getResponseCode to actually complete the connection but test doesn't care about the response
+        // We just check the logs for the expected new endpoint added or not
+        HttpUtils.getHttpConnectionWithAnyResponseCode(LS, "/basic/addEndpoint").getResponseCode();
+        
+        // This test causes SRVE0777E: Exception thrown by application class ServerContainerExt.addEndpoint which is expected
+        // Also cause SRVE0315E An exception occurred: java.lang.Throwable: java.lang.IllegalStateException: endpoint.addsclosed also expected
+        // and SRVE8094W WARNING: Cannot set header. Response already committed to show up
+        if(RepeatTestFilter.isRepeatActionActive(JakartaEE10Action.ID)){
```

Comment:
Any reason you didn't just do `JakartaEE10Action.isActive()` ?

Reply:
No specific reason, I was working with the RepeatTestFilter lately and just went with it here. I can change it back to make it consistent across all the tests

Reply:
Changed so that it matches the other tests we've worked on